agent/containermetadata/types.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+
+	docker "github.com/fsouza/go-dockerclient"
+)
+
+// DockerMetadataClient is a wrapper for the docker interface functions we need
+// We use this as a dummy type to be able to pass in engine.DockerClient to
+// our functions without creating import cycles
+// We make it exported because we need to use it for testing (Using the MockDockerClient
+// in engine package leads to import cycles)
+// The problems described above are indications engine.DockerClient needs to be moved
+// outside the engine package
+type DockerMetadataClient interface {
+	InspectContainer(string, time.Duration) (*docker.Container, error)
+}
+
+// Network is a struct that keeps track of metadata of a network interface
+type Network struct {
+	NetworkMode string `json:"NetworkMode,omitempty"`
+	IPv4Address string `json:"IPv4Address,omitempty"`
```

Comment:
Can you change this to `IPv4Addresses` instead? For now it can be a list of length 1. It'd be helpful to surface this info when we start containers with multiple IP addresses.